sdk-extensions/logging/src/main/java/io/opentelemetry/sdk/logging/data/Body.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.logging.data;
+
+import com.google.auto.value.AutoValue;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * A class that represents all the possible values for a log message body. An {@code AnyValue} can
+ * currently only have 1 types of values: {@code String}, represented through {@code AnyValue.Type}.
+ * This class will likely be extended in the future to include additional body types supported by
+ * the OpenTelemetry log data model.
+ */
+@Immutable
+public abstract class Body {
+
+  /** An enum that represents all the possible value types for an {@code AnyValue}. */
+  public enum Type {
+    STRING
+  }
+
+  /**
+   * Returns an {@code AnyValue} with a string value.
+   *
+   * @param stringValue The new value.
+   * @return an {@code AnyValue} with a string value.
+   */
+  public static Body stringBody(String stringValue) {
+    return StringBody.create(stringValue);
+  }
+
+  /** Returns the string value of this {@code Body}. */
+  public String asString() {
+    throw new UnsupportedOperationException(
```

Comment:
Since we have a string body, why not just return that. I find this exception surprising for a `Body` that just contains a string. Or maybe I'm just getting hung up on AutoValue stuff and it's handled by the `StringBody` subtype?

Reply:
there will be a bunch more types of bodies in the future. yeah, this is a little weird, but this method implementation should never actually be hit by anything, since we'll only create things as sub-classes. I'd like to see if we can convince AutoValue that this should be an interface, rather than an abstract class.

Reply:
looks like it's easy to make it an interface. Will amend.